### DISCUSSION TITLE
Is it possible to invalidate a mutation, similar to how queries are invalidated with queryClient?

### DISCUSSION THREAD

**Discussion by maxpaleo** [TanStack/query#7893] [Q&A] [answered]
Hi All, 

So I am looking for a way to invalidate the response from a mutation, similar to how a query can be invalidated using `queryClient` but haven't found anything so far. 

**Here is the situation:** 

We use state returned by the mutation to display UI elements. For example: the `isSuccess` state is used to display a check icon in a button when the update was successful. 

However, I would like automatically reset this state with a 5 second delay, so that the icon doesn't remain in the UI. 

Is there a way to invalidate this directly in the `onSuccess` of `useMutation` without having to add new local react state? 

```tsx
const {
    mutate: updateSettings,
    isPending: mutatePending,
    isSuccess, // The isSuccess state is used in the UI to show a checkmark in the action button. 
  } = useMutation({
    mutationKey: ["updateOrganizationSettings"],
    mutationFn: updateOrganizationSettings,
    ...onErrorToast(),
    onSuccess: () => {
      toast.success("Settings updated successfully"),
        queryClient.invalidateQueries({
          queryKey: ["getOrganizationById"],
        });
    },
  });
```

**Comment by TkDodo** [comment, marked as answer]
call `reset` returned from `useMutation`
